packages/ui-extensions/docs/surfaces/checkout/reference/examples/ui-components/sheet-description-preferences.example.ts

```diff
@@ -0,0 +1,96 @@
+import {
+  extension,
+  Sheet,
+  Button,
+  Link,
+  TextBlock,
+  BlockStack,
+} from '@shopify/ui-extensions/checkout';
+
+export default extension(
+  'purchase.checkout.footer.render-after',
+  (root, {customerPrivacy}) => {
+    customerPrivacy.subscribe(
+      ({shouldShowBanner}) => {
+        const primaryFragment =
+          root.createFragment();
+
+        const declineButton =
+          root.createComponent(
+            Button,
+            {
+              kind: 'secondary',
+              onPress: () => {},
+            },
+            'I decline',
+          );
```

Comment:
Same comment regarding subscribing logic where this will happen every-time the customer privacy changes.